vortex-expr/src/lib.rs

```diff
@@ -1,8 +1,50 @@
 #![feature(iter_intersperse)]
 
+use std::sync::Arc;
+
 pub mod datafusion;
 mod expr;
 mod operators;
 
 pub use expr::*;
 pub use operators::*;
+
+pub fn split_conjunction(expr: &Arc<dyn VortexExpr>) -> Vec<Arc<dyn VortexExpr>> {
+    match expr.as_any().downcast_ref::<BinaryExpr>() {
+        Some(bexp) if bexp.op() == Operator::And => {
+            let mut exprs = split_conjunction(bexp.lhs());
+            exprs.extend_from_slice(&split_conjunction(bexp.rhs()));
+            exprs
+        }
+        Some(_) | None => vec![expr.clone()],
+    }
+}
+
+pub fn expr_is_filter(expr: &Arc<dyn VortexExpr>) -> bool {
```

Comment:
This has a weird name? Column expressions are valid filters

Reply:
not sure I have a better idea 😅